httpx/_client.py

```diff
@@ -56,6 +57,12 @@
     warn_deprecated,
 )
 
+try:
+    from contextlib import asynccontextmanager  # type: ignore
+except ImportError:  # pragma: nocover
+    from async_generator import asynccontextmanager  # type: ignore
```

Comment:
Thoughts on the `_compat.py` pattern? Move this bit there and then do `from ._compat import asynccontextmanager` everywhere. Always found it ends up being worth doing eventually, and keeps compat code in a clear area.